01_fib/fib.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, fib(7))
+	}
+
+func fib(n int) int{
+	if n <=1 {
+		return n
+	}
+	return fib(n-1) + fib(n-2)
```

Comment:
i like how this decouplesibonacci sequence calculation from IO (formatting of numbers to the output stream)!